src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -0,0 +1,126 @@
+        public static ObjectManagementService Instance => objectManagementServiceInstance.Value;
+
+        public static ConnectionService connectionService;
+
+        private IProtocolEndpoint serviceHost;
+
+        public ObjectManagementService() { }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(RenameRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
+               ConnectionInfo connInfo;
+               try
+               {
+                   if (connectionService.TryFindConnection(
+                           requestParams.OwnerUri,
+                           out connInfo))
+                   {
+                       using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
+                       {
+                           IRenamable renameObject = this.GetSQLRenameObject(requestParams, sqlConn);
+
+                           renameObject.Rename(requestParams.NewName);
+                       }
+                   }
+                   else
+                   {
+                       Logger.Error("The connection could not be found.");
+                       throw new Exception(SR.ErrorConnectionNotFound);
```

Comment:
> throw new Exception(SR.ErrorConnectionNotFound); [](http://example.com/codeflow?start=22&length=49)

why is throwing a plain "Exception" considered a good practice? 
Typically, components define their own exception types so they are isolatable from other component exceptions during debugging.

Reply:
Initially I would agree with you. But I interpreted [this comment](https://github.com/microsoft/sqltoolsservice/pull/1686#discussion_r972291780), that I should use the ```Exception``` class. Should I create a separate exception or not?

Reply:
It's just not something we've ever really done here - @shueybubbles's points are valid but for now let's keep it with a plain Exception and I'll make a note to follow up and make some sort of general "ToolsServiceException" we can use instead of plain Exception objects so that we can update everything at once.